stdlib/test/builtin/test_repr.mojo

```diff
@@ -47,7 +47,24 @@ def test_none_representation():
     assert_equal(repr(None), "None")
 
 
+def test_escape_representation():
+    assert_equal(repr("\0"), r"'\x00'")
+    assert_equal(repr("\x06"), r"'\x06'")
+    assert_equal(repr("\x09"), r"'\t'")
+    assert_equal(repr("\n"), r"'\n'")
+    assert_equal(repr("\x0d"), r"'\r'")
+    assert_equal(repr("\x0e"), r"'\x0e'")
+    assert_equal(repr("\x1f"), r"'\x1f'")
+    assert_equal(repr(" "), "' '")
+    assert_equal(repr("'"), '"\'"')
+    assert_equal(repr("A"), "'A'")
+    assert_equal(repr("\\"), r"'\\'")
+    assert_equal(repr("~"), "'~'")
+    assert_equal(repr("\x7f"), r"'\x7f'")
```

Comment:
Can you please ensure that these use `String.__repr__`? Due to how implicit conversion work, I think this actually uses `StringLiteral.__repr__` as it stands right now.

Reply:
`StringLiteral.__repr__`, converts it to a String, then calls `String.__repr__` so there's not much reason to change it, unless we want to do something else for `StringLiteral.__repr__()`.
Should I use `String(..)` anyways?

Reply:
Yeah, you're right, which kind of goes to the point that we should ensure that this does not exercise `StringLiteral.__repr__` (because tests for those should be in another file, similarly enforcing that it does not use `String.__repr__`).

Reply:
(I'll do what I mentioned in the above suggestion.)